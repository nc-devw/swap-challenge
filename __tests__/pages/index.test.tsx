import { render, waitFor } from "@testing-library/react";
import Home, { getServerSideProps } from "@/pages/index";

jest.mock("../../helpers/fetchData", () => () => ({ data: [] }));

describe("Home", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<Home coins={[]} />);
    expect(asFragment).toMatchSnapshot();
  });

  it("should receive properties correctly", async () => {
    const props = await getServerSideProps();
    expect(props).toEqual({
      props: { coins: [] },
    });
  });
});
