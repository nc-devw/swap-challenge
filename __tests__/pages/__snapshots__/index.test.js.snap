// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<DocumentFragment>
  <div
    class="container mx-auto"
  >
    <div
      class="rounded-lg m-12 border shadow bg-white undefined"
    >
      <div
        class="flex flex-col"
      >
        <div
          class="text-center p-4 shadow"
        >
          <p
            class="uppercase font-bold text-lg"
          >
            Your assets
          </p>
          <p
            class="uppercase text-sm text-gray-400"
          >
            Total Balance: 0 USD
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
